sdk/iot/core/inc/az_iot_provisioning_client.h

```diff
@@ -0,0 +1,310 @@
+    az_span* out_mqtt_password);
+
+/**
+ *
+ * Register APIs
+ *
+ *   Use the following APIs when the Shared Access Key is available to the application or stored
+ *   within a Hardware Security Module. The APIs are not necessary if X509 Client Certificate
+ *   Authentication is used.
+ */
+
+/**
+ * @brief Azure IoT Provisioning status codes.
+ *
+ */
+typedef enum
+{
+  // Service success codes
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_OK = 200,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_ACCEPTED = 202,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_NO_CONTENT = 204,
+
+  // Service error codes
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_BAD_REQUEST = 400,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_UNAUTHORIZED = 401,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_FORBIDDEN = 403,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_NOT_FOUND = 404,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_NOT_ALLOWED = 405,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_NOT_CONFLICT = 409,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_PRECONDITION_FAILED = 412,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_REQUEST_TOO_LARGE = 413,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_UNSUPPORTED_TYPE = 415,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_THROTTLED = 429,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_CLIENT_CLOSED = 499,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_SERVER_ERROR = 500,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_BAD_GATEWAY = 502,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_SERVICE_UNAVAILABLE = 503,
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_TIMEOUT = 504,
+} az_iot_provisioning_client_status;
+
+/**
+ * @brief Checks if the status indicates a successful operation.
+ *
+ * @param status The #az_iot_hub_client_status to verify.
+ * @return True if the status indicates success. False otherwise.
+ */
+AZ_NODISCARD bool az_iot_provisioning_client_is_success_status(
+    az_iot_provisioning_client_status status);
+
+/**
+ * @brief Checks if the status indicates a retriable error occurred during the operation.
+ *
+ * @param status The #az_iot_hub_client_status to verify.
+ * @return True if the operation should be retried. False otherwise.
+ */
+AZ_NODISCARD bool az_iot_provisioning_client_is_retriable_status(
+    az_iot_provisioning_client_status status);
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to register responses.
+ *
+ * @param client
+ * @param mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
+ *                          filter.
+ * @param out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_provisioning_client_register_subscribe_topic_filter_get(
+    az_iot_provisioning_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief The registration operation state.
+ * @note This is returned only when the operation completed.
+ *
+ */
+typedef struct az_iot_provisioning_client_registration_state
+{
+  az_span assigned_hub_hostname; /** @brief Assigned Azure IoT Hub hostname. @note This is only

```

Comment:
Note on the comment style in my previous comment:
```
/**< ......  */
```
There should be an angle bracket after the second asterisk and then closing at the end. Also no need for the `@brief` tag.

Reply:
Ah! Way better.

Reply:
Actually FYI: mine didn't work. Changed all to /**< and tested with actual doxygen.